src/main/java/net/openhft/chronicle/core/io/TracingReferenceCounted.java

```diff
@@ -127,6 +132,37 @@ public void release(ReferenceOwner id) throws IllegalStateException {
         }
     }
 
+    @Override
+    public void reserveTransfer(ReferenceOwner from, ReferenceOwner to) throws IllegalStateException {
+        synchronized (references) {
+            final StackTrace stackTrace = references.get(to);
+            if (stackTrace != null) {
+                throw new IllegalStateException(type.getName() + " already reserved resource by " + asString(to) + " here", stackTrace);
+            }
+            if (references.remove(from) == null) {
+                throw throwInvalidReleaseException(from);
+            }
+            releases.put(from, stackTrace("reserveTransfer", from));
+            references.put(to, stackTrace("reserveTransfer", to));
+            releases.remove(to);
+        }
+        referenceChangeListeners.notifyTransferred(from, to);
+    }
```

Comment:
Had to add this because it was using the default of `reserve(to);release(from)` previously, but we want to fire the specific event of a transfer